src/os/bluestore/BlueStore.cc

```diff
@@ -2305,7 +2305,7 @@ BlueStore::OldExtent* BlueStore::OldExtent::create(CollectionRef c,
 						   BlobRef& b) {
   OldExtent* oe = new OldExtent(lo, o, l, b);
   b->put_ref(c.get(), o, l, &(oe->r));
-  oe->blob_empty = b->get_referenced_bytes() == 0;
+  oe->blob_empty = !b->is_referenced();
   return oe;
```

Comment:
Offtopic: is it possible that this change will terminate appearance of zombie spanning blobs?

Reply:
hopefully this is "safe" from this point of view. Just a bit more effective.